dbsnp-importer/src/main/java/uk/ac/ebi/eva/dbsnpimporter/sequence/FastaSequenceReader.java

```diff
@@ -16,29 +16,42 @@
 package uk.ac.ebi.eva.dbsnpimporter.sequence;
 
 import htsjdk.samtools.SAMException;
+import htsjdk.samtools.SAMSequenceDictionary;
 import htsjdk.samtools.reference.ReferenceSequenceFile;
 import htsjdk.samtools.reference.ReferenceSequenceFileFactory;
 
 import java.nio.file.Path;
 
 /**
- * Implementation of SequenceReader for indexed fasta files
+ * Class used to read regions from a given FASTA file
  */
-public class FastaSequenceReader implements SequenceReader {
+public class FastaSequenceReader {
 
     private final ReferenceSequenceFile fastaSequenceFile;
 
+    private final SAMSequenceDictionary sequenceDictionary;
+
     public FastaSequenceReader(Path fastaFile) {
         fastaSequenceFile = ReferenceSequenceFileFactory.getReferenceSequenceFile(fastaFile, true);
+        sequenceDictionary = fastaSequenceFile.getSequenceDictionary();
+        if (sequenceDictionary == null || !fastaSequenceFile.isIndexed()) {
+            throw new IllegalArgumentException("A sequence dictionary file and a Fasta index file are required");
```

Comment:
Isn't the sequence dictionary file "embedded" in the FASTA file? If that is the case I would suggest rewording as:

```
A FASTA file containing a sequence dictionary and a FASTA index file are required
```